# Work on #2017Elyséefr data

imports :

In [1]:
import pandas as pd

import datetime

#Twitter intelligence tool to scrap data from twitter : 
#https://github.com/twintproject/twint
import twint

#Pour gérer une erreur : 
#Voir https://github.com/twintproject/twint/issues/166
import nest_asyncio
nest_asyncio.apply()

# Reading and describing data :

Les fichiers sont en .tab donc il faut ajuster les paramètres de `pd.read_csv` :

In [2]:
dfprofiles=pd.read_csv('Donnees/data_elysee2017_tweets/profiles_annotations.tab', sep='\t')

In [3]:
dfprofiles.head()

FROM_USER_ID PROFILE_NATURE PARTY  MEDIA_PROFESSIONAL SEX
0    1264925822     individual    fi                 NaN   f
1    1265204928     individual    fi                 NaN   f
2    1265310602     individual    fi                 NaN   m
3    1265661672     individual    fi                 NaN   m
4    1267543572     individual    fi                 NaN   m

In [4]:
dfprofiles.shape

(22853, 5)

#### Données sur le parti politique : 

In [5]:
dfprofiles['PARTY'].isna().sum()

3401

Il n'y a que 3401 nans pour le parti politique sur les 22853 profils disponibles. 

Etant donné qu'il serait très compliqué de les estimer (sauf à la main) donc on peut les supprimer. 

In [6]:
dfprofiles.dropna(0,subset=['PARTY'],inplace=True)

#### Autres données : 

In [7]:
dfprofiles.MEDIA_PROFESSIONAL.unique()

array([nan,  1.])

On remplace les nan dans `MEDIA_PROFESSIONAL` par 0 
- S'il s'agit d'un media professionel : 1
- S'il s'agit d'un individu : 0

In [8]:
dfprofiles['MEDIA_PROFESSIONAL'].fillna(0,inplace=True)

In [9]:
print('nombre de medias dans le base : ', dfprofiles[dfprofiles['MEDIA_PROFESSIONAL']==1].shape[0])
print("nombre d'individus"," dans le base : ", dfprofiles[dfprofiles['MEDIA_PROFESSIONAL']==0].shape[0])

nombre de medias dans le base :  46
nombre d'individus  dans le base :  19406


**Il y a beaucoup plus d'invidus que de media, ce qui est rassurant compte tenu de l'utilisation que l'on veut faire**

In [10]:
dfprofiles.isna().sum()

FROM_USER_ID             0
PROFILE_NATURE           0
PARTY                    0
MEDIA_PROFESSIONAL       0
SEX                   5338
dtype: int64

Il reste 5338 nans dans les données. 
**On ne s'en occupe pas pour l'instant**

#### Nombre d'utilisateurs uniques  : 

In [11]:
dfprofiles['FROM_USER_ID'].nunique()

19452

# Scraping :  

In [2]:
user_list=[1264925822]
#Moi : 
#user_list=[796492866]

In [3]:
#Boucle for sur tous les ids utilisateurs de la base : 
for u_id in user_list: 
    filename='Donnees/tweets_pol/'+str(u_id)+'.csv'
    #Lancer la configuration : 
    c = twint.Config()
    #On défini l'user : 
    c.User_id=u_id
    #On ne cherche que sur 2017 : 
    
    c.Year=2017
    
    #c.Since='2015-12-20'
    #c.Until="2017-12-28"
    
    #Supprimer cette ligne pour tout récupérer :
    #c.Custom["tweet", "nlikes"] = ["tweet", "nlikes"]
    
    c.Output = filename
    #On stock dans un csv :
    c.Store_csv = True
    #Nous ne souhaitons pas voir les tweets apparaitre dans la console Python
    c.Hide_output = True  
    #Nous ne récupérons pas les hashtags
    c.Show_hashtags = False   
    
    
    
    #c.Store_object = True # nous stockons également les tweets dans un objet
    #On lance la recherche :
    twint.run.Search(c)
    #Tous les tweets seront conservés dans la variable tweets_as_objects
    #tweets_as_objects = twint.output.tweets_object 

CRITICAL:root:twint.run:Twint:Feed:noDataExpecting value: line 1 column 1 (char 0)
CRITICAL:root:twint.run:Twint:Feed:noDataExpecting value: line 1 column 1 (char 0)
CRITICAL:root:twint.run:Twint:Feed:noDataExpecting value: line 1 column 1 (char 0)
CRITICAL:root:twint.run:Twint:Feed:noDataExpecting value: line 1 column 1 (char 0)
CRITICAL:root:twint.run:Twint:Feed:noDataExpecting value: line 1 column 1 (char 0)
CRITICAL:root:twint.run:Twint:Feed:noDataExpecting value: line 1 column 1 (char 0)
CRITICAL:root:twint.run:Twint:Feed:noDataExpecting value: line 1 column 1 (char 0)
CRITICAL:root:twint.run:Twint:Feed:noDataExpecting value: line 1 column 1 (char 0)
CRITICAL:root:twint.run:Twint:Feed:noDataExpecting value: line 1 column 1 (char 0)
CRITICAL:root:twint.run:Twint:Feed:noDataExpecting value: line 1 column 1 (char 0)
CRITICAL:root:twint.run:Twint:Feed:noDataExpecting value: line 1 column 1 (char 0)
CRITICAL:root:twint.run:Twint:Feed:noDataExpecting value: line 1 column 1 (char 0)
CRIT

Expecting value: line 1 column 1 (char 0) [x] run.Feed
[!] if get this error but you know for sure that more tweets exist, please open an issue and we will investigate it!


In [4]:
#help(twint.output)

In [5]:
#help(twint.Config)

In [6]:
df=pd.read_csv('Donnees/tweets_pol/'+str(u_id)+'.csv', error_bad_lines=False)

In [7]:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', 100)
pd.set_option('display.max_colwidth', 500)

In [8]:
df

id     conversation_id     created_at        date  \
0     814994099262652417  814990778703319040  1483144709000  2016-12-31   
1     814993901262139392  814991711952928768  1483144662000  2016-12-31   
2     814638683504734208  813865398022062081  1483059972000  2016-12-30   
3     814637636593848320  814438822582845440  1483059722000  2016-12-30   
4     812013763129917440  812003707546112000  1482434142000  2016-12-22   
5     812003934466441216  811994668691058688  1482431798000  2016-12-22   
6     812003928367923200  811958435508088832  1482431797000  2016-12-22   
7     812003920767844358  811983087911534592  1482431795000  2016-12-22   
8     812003707546112000  812003707546112000  1482431744000  2016-12-22   
9     812001754808250368  812001754808250368  1482431279000  2016-12-22   
10    811650983675641856  811235998344548352  1482347648000  2016-12-21   
11    811650417381679104  811506379613736961  1482347513000  2016-12-21   
12    811617422172160001  811298163407671296  1482339647000  2016-12-21   
13    810847031988195328  810847031988195328  1482155971000  2016-12-19   
14    810697144952823808  810697144952823808  1482120236000  2016-12-19   
15    810650486584000514  810470399624511488  1482109111000  2016-12-19   
16    810629718588477440  810470399624511488  1482104160000  2016-12-19   
17    810628503708246016  810536634601107460  1482103870000  2016-12-19   
18    809537285641437184  809537285641437184  1481843704000  2016-12-16   
19    809535911100903424  809406770166792192  1481843376000  2016-12-16   
20    809534167541235718  809422567144587264  1481842960000  2016-12-16   
21    809533667697692673  809435684696195075  1481842841000  2016-12-16   
22    809519411199701000  809519411199701000  1481839442000  2016-12-15   
23    809137598262374400  793559457322205184  1481748411000  2016-12-14   
24    809090815591071750  809080333266259968  1481737257000  2016-12-14   
25    808741616588652553  808741616588652553  1481654001000  2016-12-13   
26    808712622283649024  808649543474704384  1481647088000  2016-12-13   
27    808391636640301057  808388487527497729  1481570560000  2016-12-12   
28    808365141960048640  808321101117976580  1481564243000  2016-12-12   
29    808091305872080897  808082170858315776  1481498955000  2016-12-12   
30    808063482583273475  808059234323759104  1481492321000  2016-12-11   
31    808032116718104576  808032116718104576  1481484843000  2016-12-11   
32    808025073554690048  808025073554690048  1481483164000  2016-12-11   
33    807972096747982848  807796542031728640  1481470533000  2016-12-11   
34    807753800278802432  807751786907963393  1481418487000  2016-12-11   
35    807702275766571008  807702275766571008  1481406203000  2016-12-10   
36    807688142601981953  797820208433995777  1481402833000  2016-12-10   
37    807680378811150336  807680378811150336  1481400982000  2016-12-10   
38    807636453270253568  807548496572190720  1481390510000  2016-12-10   
39    807635645120139264  807555809517432832  1481390317000  2016-12-10   
40    807403828828061696  807403828828061696  1481335048000  2016-12-10   
41    807376400399351808  807324526619099136  1481328508000  2016-12-10   
42    807375257904545797  807375257904545797  1481328236000  2016-12-10   
43    807281211999657984  807281211999657984  1481305814000  2016-12-09   
44    807276525108293633  807271540115042304  1481304696000  2016-12-09   
45    807269019912962048  807269019912962048  1481302907000  2016-12-09   
46    806983899553427461  806962626962092038  1481234929000  2016-12-08   
47    806541728354332674  806460525920010240  1481129507000  2016-12-07   
48    806490488761094144  806460525920010240  1481117291000  2016-12-07   
49    806429597642485760  806429597642485760  1481102773000  2016-12-07   
...                  ...                 ...            ...         ...   
2370  483670744220856321  483670744220856321  1404151066000  2014-06-30   
2371  483663145769381888  483663145769381888  1

### code d'exemple : 

In [ ]:
for compte in compte_twitter:
 #nom du fichier csv qui sera produit
     filename = compte+".csv"
 #lancement de l’objet twint
     c = twint.Config()
 #nom du compte que nous recherchons
     c.Username = compte
 #Nous désirons stocker un fichier csv contenant l’ensemble des tweets récupérés
     c.Store_csv = True
     # Nombre de tweets récupérés
     c.Count = True
 #Si nous avons déjà un fichier csv des tweets, nous n’analysons pas le compte Tweeter de la personnalité
     if os.path.exists(filename):
         print('déjà présent :' + compte )
 #Sinon, nous récuperons la liste des tweets
     else:
         c.Custom["tweet"] = ["tweet"]
         c.Output = filename
 #Nous ne souhaitons pas voir les tweets apparaitre dans la console Python
         c.Hide_output = True  
 #Nous ne récupérons pas les hashtags
         c.Show_hashtags = False     
         c.Store_object = True # nous stockons également les tweets dans un objet  
 #Lancement de la recherche
         twint.run.Search(c)
 #Tous les tweets seront conservés dans la variable tweets_as_objects
         tweets_as_objects = twint.output.tweets_object 

Erreur :

In [ ]:
#Same error and just following the simple code of query search and save it in a panda.

import twint
import pandas as pd
c = twint.Config()
c.Search = "agroecología"
c.Pandas = True
twint.run.Search(c)
Tweets_df = twint.storage.panda.Tweets_df
Tweets_df.to_csv(r'Query2ES.csv', index = None, header=True)
twint.storage.panda.clean

## From : https://towardsdatascience.com/analyzing-tweets-with-nlp-in-minutes-with-spark-optimus-and-twint-a0c96084995f

In [ ]:
c=twint.Config()
#c.Search = "data"
c.User_id=1264925822
# Custom output format
#c.Format = "Username: {username} |  Tweet: {tweet}"
c.Limit = 1
c.Year=2017
c.Pandas = True

twint.run.Search(c)
Tweets_df = twint.storage.panda.Tweets_df

In [ ]:
Tweets_df.head()

In [ ]:
def available_columns():
    return twint.output.panda.Tweets_df.columns

In [ ]:
available_columns()

In [ ]:
username='adrienhans'

In [1]:
import twint

c = twint.Config()
c.Limit = 20
c.Username = username
c.Pandas = True
twint.run.Search(c)
Tweets_df = twint.storage.panda.Tweets_df

NameError: name 'username' is not defined

In [ ]:
help(twint.run.Search)

# Other data : 

In [2]:
import pandas as pd

In [3]:
df_fi_tweets_id=pd.read_csv('Donnees/data_elysee2017_tweets/posts_ids_fi.tab', sep='\t')

In [5]:
df_fi_tweets_id.shape

(2192544, 1)

In [5]:
import twint

In [10]:
#c.User_full = True
help(twint.Config().User_id)

Help on NoneType object:

class NoneType(object)
 |  Methods defined here:
 |  
 |  __bool__(self, /)
 |      self != 0
 |  
 |  __repr__(self, /)
 |      Return repr(self).
 |  
 |  ----------------------------------------------------------------------
 |  Static methods defined here:
 |  
 |  __new__(*args, **kwargs) from builtins.type
 |      Create and return a new object.  See help(type) for accurate signature.

